File: neetcode/blind75/1D_DP/climbing_stairs.py
# https://neetcode.io/problems/climbing-stairs

class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 3:
            return n;

        dp_table = [0] * (n+1);
        dp_table[1] = 1
        dp_table[2] = 2
        dp_table[3] = 3

        for i in range(4, n+1):
            dp_table[i] = dp_table[i-1] + dp_table[i-2]

        return dp_table[n];

File: neetcode/neet150_JS/1-D-Dynamic-Programming/climbing-stairs.js
// https://neetcode.io/problems/climbing-stairs

class Solution {
    /**
     * @param {number} n
     * @return {number}
     */
    climbStairs(n) {
        if (n <= 2) {
            return n
        }

        const dpTable = Array(n + 1).fill(0)
        dpTable[1] = 1
        dpTable[2] = 2

        for (let i = 3; i < n + 1; i ++) {
            dpTable[i] = dpTable[i - 1] + dpTable[i - 2]
        }

        return dpTable[n]
    }
}
